lib/locater.js

```diff
@@ -7,52 +7,56 @@
 function findIndices(target, line) {
   var indices = [];
 
-  if (typeof target === 'string') {
-    var index = 0;
-    var offset = 0;
+  //converts all targets to regex
+  if (!(target instanceof RegExp)) {
+    target = new RegExp('\\b' + target + '\\b', "g");
+  }
 
-    while ((index = line.indexOf(target, offset)) !== -1) {
-      indices.push(index);
-      offset = index + 1;
-    }
-  } else if (target instanceof RegExp) {
-    while ((match =  target.exec(line)) !== null) {
-      target.lastIndex = match.index + 1;
-      indices.push(match.index);
-    }
+  while ((match =  target.exec(line)) !== null) {
+    indices.push(match.index);
   }
 
   return indices;
 }
 
+/**
+ * Removes the last line break from a string.
+ * @param line {String} - the string to remove line break from
+ */
+function removeLastLineBreaks(source) {
+  return source.replace(/[\n]+$/, '')
+}
+
+/**
+ * Converts sting to lines based on \n. Also removes last line break
+ * @param line {String} - the string split
+ */
+function splitIntoLines(source) {
+  return source.split('\n');
+}
+
 module.exports = {
   find: function (target, source) {
-    var sourceLines = source.split('\n');
-    var lineNumber = 1;
+    var sourceLines = splitIntoLines( removeLastLineBreaks(source) );
     var results = [];
 
-    sourceLines.forEach(function (line, idx) {
-      // Do not count the last line that is empty
-      if (idx === sourceLines.length - 1 && ! line) {
-        return;
-      }
-
-      var indices = findIndices(target, line);
-
-      indices.forEach(function (index) {
-        var position = {line: lineNumber, cursor: index + 1};
```

Comment:
I thought we might be able to convert all instances of target to array so we can treat it the same for all input and run it through the forEach saving on if else statements.